qutebrowser/browser/hints.py

```diff
@@ -833,7 +833,7 @@ def filter_hints(self, filterstr):
         for elems in self._context.elems.values():
             try:
                 if (filterstr is None or
-                        str(elems.elem).lower().startswith(filterstr)):
+                        filterstr.lower() in str(elems.elem).lower()):
```

Comment:
Case-insensitive matching would be a good idea IMHO, and it also helps with #577 (filtering hints by typing uppercase text even in letter hinting mode).

What about adding a new option, say `hints -> case-insensitive-filtering` or so? That's the usual thing I do when I can imagine people might have different preferences :)